### ISSUE TITLE
[Feat] Log Flops to wandb using callback

### ISSUE THREAD

**Issue by NanoCode012** [axolotl-ai-cloud/axolotl#564]
### ⚠️ Please check that this feature request hasn't been suggested before.

- [X] I searched previous [Ideas in Discussions](https://github.com/OpenAccess-AI-Collective/axolotl/discussions/categories/ideas) didn't find any similar feature requests.
- [X] I searched previous [Issues](https://github.com/OpenAccess-AI-Collective/axolotl/labels/enhancement) didn't find any similar feature requests.

### 🔖 Feature description

Logging Flops to wandb would be good to see performance.

### ✔️ Solution

Flops logged to wandb.

### 📝 Additional Context

A user on discord mentioned that it can be found within the following:

> self.state.total_flos -> https://github.com/huggingface/transformers/blob/7ccac73f749ce535851b9188f3867d5ed87c318c/src/transformers/trainer.py#L675C84-L675C105

We can create a callback to log this to wandb.

Ref: https://discord.com/channels/1104757954588196865/1110594519226925137/1151530888757260338

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this feature has not been requested yet.
- [X] I have provided enough information for the maintainers to understand and evaluate this request.

**Comment by mhenrichsen** [comment]
@NanoCode012 is this still something you would want implemented?

**Comment by NanoCode012** [comment]
Yeah, I think this can be a quick callback to add though I haven't verified `flos` refers to the FLOPS

**Comment by NanoCode012** [comment]
I went and checked that `total_flos` is the FLOPS count, however, the number may be off (GH Issue about miscounting for embed layers). Given that it may be incorrect, I'm not sure we should add it.
